types/logger/logger.go

```diff
@@ -35,6 +36,23 @@ type Context context.Context
 
 type logfKey struct{}
 
+// J marshals v as JSON and writes it to logf formatted with the annotation to make logtail
+// treat it as a structured log.
+func J(logf Logf, recType string, v interface{}) {
+	// TODO: pool of json.Encoder+bytes.Buffer pre-wired pairs
+	enc, err := json.Marshal(struct {
```

Comment:
There's a benefit to encoding the type as a key in the JSON object, rather than as a member field. This ensures that the type has to appear before the payload at a syntactic level. One of my frustrations implementing `google.protobuf.Any` in JSON is that we could not always know the type before we parsed the body, which made a few optimizations difficult.

Thus, something like:
```go
enc, err := json.Marshal(val)
if err != nil {
    ...
}
logf("[v\x00JSON:{%q:%s}", recType, enc)
```

(I'm making the assumption that `recType` contains no special characters such that "%q" is equivalent to a JSON serialized string)